src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
TBH, I'm not sure if this kind of state would even need to be migrated into a store/reducer in a Flux/Redux architecture.

I If we were building a Redux app, we would still want to gradually migrate state from leaf components up into components closer to the trunk and then eventually into a reducer. But it's not a given that the state will always end up in a reducer -- each migration is a conscious decision we'd make, backed by a compelling reason.

So the bottom line is that I think we might be thinking ahead too far. Let's take things one step at a time and strive for simplicity in the code.